R/processing_pcs.R

```diff
@@ -0,0 +1,65 @@
+#' Remove cells with NA batch label from PCs and label rownames
+#'
+#' @param pcs The PCs to be filtered and labeled
+#' @param batches Vector of cell-wise batch information whose length is the number of
+#'   rows in `pcs`
+#'
+#' @return PCs with NA batch cells removed and labeled rownames
+filter_pcs <- function(pcs, batches) {
+
+  # Remove NA batch cells
+  retain_indices <- which(!is.na(batches))
+
+  # Check that there are retained indices present
+  if(length(retain_indices) == 0) {
+    stop("There are no batch cells that are not NA present.")
+  }
+
+  # Filter batches and PCs
+  batches <- batches[retain_indices]
+  pcs <- pcs[retain_indices, ]
+
+  # Check dimensions after filtering
+  if (nrow(pcs) != length(batches)) {
+    stop("Incompatable PC and batch information dimensions after removing NAs.")
+  }
+
+  # Set PC rownames to be the batches
+  rownames(pcs) <- batches
+
+  return(pcs)
+}
+
+
+
+#' Downsample PCs for use in integration metric calculations
+#'
+#' @param pcs The PCs to downsample, these PCs should contain batch labels as rownames
+#' @param frac_cells The fraction of cells to downsample to
+#'
+#' @return List with two items: `pcs`, the downsampled PCs; `batch_labels`, the
+#'  corresponding batch labels for the downsampled PCs
+downsample_pcs <- function(pcs, frac_cells) {
+
+  # Check that there is a minimum number of cells
+  num_cells <- nrow(pcs)
+  if (!num_cells > 1) {
```

Comment:
Do we want to increase this? I feel like 1 is really low but I don't know what a good threshold is? 50? 25?

Reply:
I am also not sure of a good threshold, I implemented 50 for now unless @jashapiro has objections.